File: BinaryTreeLevelOrderTraversal/fifth.py
def levelOrder(self, root):
    res,next=[],[]
    if root:
        temp=[root]
    else:
        return res
    res.append(temp)
    while 1:
        for v in temp:
            if v.left:
                next.append(v.left)                
            if v.right:
                next.append(v.right)
        if next==[]:
            break
        res.append(next)
        temp=list(next)
        next=[]
    return [[v.val for v in x] for x in res]

File: BinaryTreeLevelOrderTraversal/fifth.java
public List<List<Integer>> levelOrder(TreeNode root) {
    List<List<Integer>> res = new ArrayList<>();
    if (root == null) {return res;}
    Queue<TreeNode> q = new LinkedList<>();
    Queue<Integer> level = new LinkedList<>();
    q.offer(root);
    level.offer(1);
    while (!q.isEmpty()) {
        TreeNode temp = q.poll();
        int tempLevel = level.poll();
        if (res.size() >= tempLevel) {res.get(tempLevel - 1).add(temp.val);}
        else {
            List<Integer> per = new ArrayList<>();
            per.add(temp.val);
            res.add(per);
        }
        if (temp.left != null) {q.offer(temp.left);level.offer(tempLevel + 1);}
        if (temp.right != null) {q.offer(temp.right);level.offer(tempLevel + 1);}
    }
    return res;
}
The idea is: since the question is related to the level, I think about the bfs firstly. Using a queue to store the TreeNode, and using another queue to store the level number. Once the number of level is smaller or equal to the size of result, which is the list, add the node's value to the existed list. Otherwise, create a new list, which means a new level.
